bcrypt.js

```diff
@@ -175,3 +190,54 @@ module.exports.getRounds = function(hash) {
 
     return bindings.get_rounds(hash);
 };
+
+/// convert a node style callback function
+/// to one returning a promise
+/// @param {f} function to be encapsulated
+/// @param {args} Array like, args to be passed to the called function
+/// @return {Promise} a promise encapuslaing the function
+function _promise(f,args) {
+    //cant do anything without promise
+
+    if(typeof Promise === 'undefined')
+        return;
+    //just polyfill here (for 0.12)
+    //this just wont work on node 0.12
+    //var _from = Array.from || [].slice.call;    
+
+    //for node 0.12 stupidity
+    var _from = Array.from;
+    if(!_from)
+    _from = function(args) {
+        return [].slice.call(args);    
+    };
+    
+    if(typeof args !== 'Array')
+        args = _from(args);
```

Comment:
I created a eslint code style to help me adhere to the style of the rest of the code. 
Here is it: https://gist.github.com/Agathver/9762cc1c65878753e4e799348ee0411c